x/lockup/keeper/lock.go

```diff
@@ -722,3 +722,78 @@ func (k Keeper) getCoinsFromLocks(locks []types.PeriodLock) sdk.Coins {
 	}
 	return coins
 }
+
+func (k Keeper) RebondTokens(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, coins sdk.Coins) error {
+	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return err
+	}
+
+	if lock.Owner != owner.String() {
+		return fmt.Errorf("lock %d is not owned by %s", lockID, owner)
+	}
+
+	if !lock.IsUnlocking() {
+		return fmt.Errorf("lock %d is not unlocking, rebonding only possible in unlocking stage", lockID)
+	}
+
+	// If all checks pass, we can rebond the tokens
+	err = k.rebondTokens(ctx, owner, *lock, coins)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
What is the reason for having 2 methods? I think we only need one unexported method

Reply:
was following the convention of [this](https://github.com/osmosis-labs/osmosis/blob/main/x/lockup/keeper/lock.go#L164-L177) example. Also, it made sense to me to separate the function that asserts the provided parameters are valid from the function that actually executes the rebonding logic. wdyt, should we do it in one function?